frontend/app/src/components/history/ClosedTrades.vue

```diff
@@ -242,360 +307,357 @@ enum TradeFilterKeys {
-    }
-  }
 
-  get allSelected(): boolean {
-    const strings = this.data.map(({ tradeId }) => tradeId);
-    return (
-      strings.length > 0 && isEqual(sortBy(strings), sortBy(this.selected))
-    );
-  }
+      tradeToDelete.value = null;
+      confirmationMessage.value = '';
+      fetch();
+    };
 
-  async ignoreTrades(ignore: boolean) {
-    let status: ActionStatus;
-
-    const actionIds = this.data
-      .filter(({ tradeId, ignoredInAccounting }) => {
-        return (
-          (ignore ? !ignoredInAccounting : ignoredInAccounting) &&
-          this.selected.includes(tradeId)
-        );
-      })
-      .map(({ tradeId }) => tradeId)
-      .filter((value, index, array) => array.indexOf(value) === index);
-
-    if (actionIds.length === 0) {
-      const choice = ignore ? 1 : 2;
-      this.setMessage({
-        success: false,
-        title: this.$tc('closed_trades.ignore.no_actions.title', choice),
-        description: this.$tc(
-          'closed_trades.ignore.no_actions.description',
-          choice
-        )
-      });
-      return;
-    }
-    const payload: IgnoreActionPayload = {
-      actionIds: actionIds,
-      type: IgnoreActionType.TRADES
+    const clearDialog = () => {
+      const formVal = form.value as any;
+      formVal?.reset();
+
+      openDialog.value = false;
+      editableItem.value = null;
     };
-    if (ignore) {
-      status = await this.ignoreActions(payload);
-    } else {
-      status = await this.unignoreActions(payload);
-    }
 
-    if (status.success) {
-      const total = this.selected.length;
-      for (let i = 0; i < total; i++) {
-        this.selected.pop();
+    const save = async () => {
+      const formVal = form.value as any;
+      if (formVal) {
+        const success = await formVal.save();
+        if (success) {
+          clearDialog();
+        }
       }
-    }
-  }
-
-  @Emit()
-  refresh() {}
+    };
 
-  @Prop({ required: true })
-  data!: TradeEntry[];
+    const isSelected = (item: Trade) => {
+      return selected.value.includes(item.tradeId);
+    };
 
-  visibleTrades: TradeEntry[] = [];
+    const selectionChanged = (item: Trade, select: boolean) => {
+      const key = item.tradeId;
+      const selection = [...selected.value];
+      if (!select) {
+        const index = selection.indexOf(key);
+        if (index >= 0) {
+          selection.splice(index, 1);
+        }
+      } else if (key && !selection.includes(key)) {
+        selection.push(key);
+      }
 
-  mounted() {
-    this.applyFilter();
-  }
+      selected.value = selection;
+    };
 
-  updateFilter(filter: MatchedKeyword<TradeFilterKeys>) {
-    this.filter = filter;
-    this.applyFilter();
-  }
+    const isAllSelected = computed(() => {
+      const strings = trades.value.map(({ tradeId }) => tradeId);
+      return (
+        strings.length > 0 && isEqual(sortBy(strings), sortBy(selected.value))
+      );
+    });
+
+    const setAllSelected = (select: boolean) => {
```

Comment:
Is there a chance we could re-use the selection composable here? https://github.com/rotki/rotki/blob/61a297126e9cb602519ea255b25d30b350c14421/frontend/app/src/views/history/composables/selection.ts#L7

Reply:
Didn't notice this exist, worth to try........

Reply:
No worries the codebase is big, and you can see that I also sometimes forget things and re-implement them :rofl: